### ISSUE TITLE
how to improve str_acc

### ISSUE THREAD

**Issue by wuyang-dl** [tancik/StegaStamp#12]
hi, can you give me some advise about:
**1** how to improve str acc?
I use win10 + tf cpu to train a model with the same image dataset you offered.
![image](https://user-images.githubusercontent.com/55725716/80668413-6cb93b80-8ad4-11ea-99ef-46fee6611e37.png)
perhaps the iterate does not reach its maximum num(14w, now is about 1.9w)

**2** and are the parameters in parser.add_argument(in train.py) correct? I need some details about these paras, to make sure whether I am on the right way training model.

**3** can you tell me the ideal about these tf encode/decode/discriminator models, why do we need these 10 or 5 CNN layers? and why can these models achieve a good performance?

**4** and can you tell me the BCH code? in code, you use BCH_POLYNOMIAL = 137, BCH_BITS = 5. this BCH code need 127bits(92bits are valid bits that we can hide data). But in parper, you use 100bits(96bits+4bits) which can recover 56 bits data.

many thanks!

**Comment by tancik** [comment]
1. I have not tested with win10 + tf cpu. The model is designed to be trained with a gpu, using a cpu may take a very long time. To match the performance in the paper, the model needs to be trained for 140k iterations.

2. The parameters listed [here](https://github.com/tancik/StegaStamp/blob/master/scripts/base.sh)  match the paper.

3. To learn more about the design decisions take a look at the corresponding research paper [here](https://arxiv.org/abs/1904.05343)

4.  In the provided code we use 100 bits to hide 56 bits. The related code is [here](https://github.com/tancik/StegaStamp/blob/2e852eeef967aa812f2d7ab79a7b55bd4ddd9c12/encode_image.py#L52). It is possible to encode longer  messages with these BCH parameters. Given that our StegaStamps were trained to hide 100 bits, these were the best BCH parameters we could find. In retrospect we should have trained the algorithm to hide 127 bits (as you mention above) for better efficiency.

**Comment by wuyang-dl** [comment]
Thanks!
I have an older version of your paper, a newer version you offered is really helpful!
I will have a deep look at this paper.

ps.
Do you have any ideas about how to encode doc images, I find it is really difficult to hide info(I think doc image is too simple, info is not easy to decode or encode).
the accuracy of decoding doc image is really low.

Thanks again!

**Comment by tancik** [comment]
When you say "doc images" are you referring to text documents? If so I would not expect this method to perform well in this situation. The method works best with natural images where the data can be hidden in the textures.

**Comment by wuyang-dl** [comment]
ok, thanks

**Comment by zimenglan-sysu-512** [comment]
hi @tancik 
i train a modeil with 140k iterations, and find that the bit_acc can reach 80%~90%, but the str_acc only get 30%~40%, is that normal?
![image](https://user-images.githubusercontent.com/13795888/96395282-d3ebb900-11f6-11eb-9d8c-6279b3628c55.png)

**Comment by tancik** [comment]
That seems reasonable. Keep in mind that the string accuracy reported in the tensorboard does not use error correcting codes.

**Comment by zimenglan-sysu-512** [comment]
thanks @tancik 
btw, if i train a larger dataset, like mirflickr (100w), how to change the training settings? like batchsize, learning rate, steps, each ramp?

**Comment by tancik** [comment]
I haven't tried a larger dataset, but my guess is that the same parameters would work fine.

**Comment by zimenglan-sysu-512** [comment]
ok, thanks.

**Comment by zimenglan-sysu-512** [comment]
hi @tancik 
i train the mirflickr (100w) with the same settings as paper, except the step changing from 14w to 140w, the result as below:
![image](https://user-images.githubusercontent.com/13795888/96857493-9d7d9a80-1491-11eb-87a3-ed0ce51f1187.png)
it seems that training a model with larger dataset and long iterations is better.

**Comment by tancik** [comment]
Very interesting! Good to know.
